pkg/kv/kvserver/replica_lifecycle_datadriven_test.go

```diff
@@ -257,36 +243,98 @@ func TestReplicaLifecycleDataDriven(t *testing.T) {
+				splitKey := dd.ScanArg[string](t, d, "split-key")
 				rs := tc.mustGetRangeState(t, rangeID)
 				desc := rs.desc
+				require.True(
+					t,
+					roachpb.RKey(splitKey).Compare(desc.StartKey) > 0 &&
+						roachpb.RKey(splitKey).Compare(desc.EndKey) < 0,
+					"split key not within range",
+				)
+				leftDesc := desc
+				leftDesc.EndKey = roachpb.RKey(splitKey)
+				rightDesc := desc
+				rightDesc.RangeID = tc.nextRangeID
+				tc.nextRangeID++
+				rightDesc.StartKey = roachpb.RKey(splitKey)
+				split := &roachpb.SplitTrigger{
+					LeftDesc:  leftDesc,
+					RightDesc: rightDesc,
+				}
 
-				return tc.mutate(t, func(batch storage.Batch) {
-					rec := (&batcheval.MockEvalCtx{
-						ClusterSettings:        tc.st,
-						Desc:                   &desc,
-						Clock:                  tc.clock,
-						AbortSpan:              rs.abortspan,
-						LastReplicaGCTimestamp: rs.lastGCTimestamp,
-						RangeLeaseDuration:     tc.rangeLeaseDuration,
-					}).EvalContext()
-
-					in := batcheval.SplitTriggerHelperInput{
-						LeftLease:      rs.lease,
-						GCThreshold:    &rs.gcThreshold,
-						GCHint:         &rs.gcHint,
-						ReplicaVersion: rs.version,
-					}
-					_, _, err := batcheval.TestingSplitTrigger(
-						ctx, rec, batch, enginepb.MVCCStats{}, split, in, hlc.Timestamp{},
-					)
-					require.NoError(t, err)
+				// Run the split trigger evaluation and capture the batch that's
+				// generated for replication. Stash it away. This represents the
+				// raft log entry that will be applied as part of split
+				// application.
+				batch := tc.storage.NewBatch()
+				rec := (&batcheval.MockEvalCtx{
+					ClusterSettings:        tc.st,
+					Desc:                   &desc,
+					Clock:                  tc.clock,
+					AbortSpan:              rs.abortspan,
+					LastReplicaGCTimestamp: rs.lastGCTimestamp,
+					RangeLeaseDuration:     tc.rangeLeaseDuration,
+				}).EvalContext()
+
+				in := batcheval.SplitTriggerHelperInput{
+					LeftLease:      rs.lease,
+					GCThreshold:    &rs.gcThreshold,
+					GCHint:         &rs.gcHint,
+					ReplicaVersion: rs.version,
+				}
+				_, _, err := batcheval.TestingSplitTrigger(
+					ctx, rec, batch, enginepb.MVCCStats{}, split, in, hlc.Timestamp{},
+				)
+				require.NoError(t, err)
+
+				tc.splits[rangeID] = &pendingSplit{
+					trigger: split,
+					batch:   batch,
+				}
+				tc.updatePostSplitRangeState(ctx, t, batch, split)
 
-					tc.updatePostSplitRangeState(t, ctx, batch, rangeID, split)
+				output, err := print.DecodeWriteBatch(batch.Repr())
+				require.NoError(t, err)
+				return strings.ReplaceAll(output, "\n\n", "\n")
+
+			case "apply-split":
+				rangeID := dd.ScanArg[roachpb.RangeID](t, d, "range-id")
+				ps, ok := tc.splits[rangeID]
+				require.True(t, ok, "pending split not found for range-id %d", rangeID)
+				split := ps.trigger
+
+				// Determine if the RHS is "destroyed" by checking replica
+				// state. this mirrors the logic in validateandpreparesplit:
```

Comment:
Clean up the comment a bit? Wrap to 80, start sentences with a capital, spell `validateAndPrepareSplit` to match the func name.